src/common/crypto/mbedtls/cryptoprovider.cpp

```diff
@@ -374,9 +374,103 @@ aos::Error MbedTLSCryptoProvider::ParseX509Certs(mbedtls_x509_crt* currentCrt, a
         return err;
     }
 
+    err = GetX509CertPublicKey(cert, currentCrt);
+    if (!err.IsNone()) {
+        return err;
+    }
+
     return GetX509CertExtensions(cert, currentCrt);
 }
 
+aos::Error MbedTLSCryptoProvider::GetX509CertPublicKey(aos::crypto::x509::Certificate& cert, mbedtls_x509_crt* crt)
```

Comment:
Better to name `Parse`, `Convert`. I guess `mbedtls_x509_crt*` could be const? Put it as first parameter, like src,dst.  Name parameters: `x509Cert`, `mbedTLSCert`. crt, cert are ambiguous.

Reply:
Done